### DISCUSSION TITLE
How do we access the nlcst from remark-retext mutate mode?

### DISCUSSION THREAD

**Discussion by tvquizphd** [remarkjs/remark#589] [Q&A] [answered]
## Initial question

```js
var unified = require('unified')
var parse = require('remark-parse')
var english = require('retext-english')
var remark2retext = require('remark-retext')

var tree_from_md = unified().use(parse).use(remark2retext, english.Parser).parse('## Hello World')
var tree_from_text = unified().use(english).parse('Hello World')
```

I would expect both `tree_from_md` and `tree_from_text` to be identical **nlcst** objects, based on the [unified docs](https://github.com/unifiedjs/unified#processing-between-syntaxes) explanation of mutate mode:
>  it sets a new syntax tree and discards the origin tree.

However, `tree_from_md.children[0].type == 'heading'` while `tree_from_text.children[0].type == 'ParagraphNode'`. This example (among others) indicates to me that the tree produced using `remark2retext` remains an unmodified **mdast** instead of my desired output of a modified **nlcst**.

As I'm very new to unified, and unist in general, I'm sure what I'm observing is the expected behavior while I'm simply misunderstanding the documentation. So, essentially, what I'm looking for is how to retrieve an **nlcst** the output of `remark-retext`. 

## Temporary solutions

This solution requires me to include the following imports:
```js
var toNlcst = require('mdast-util-to-nlcst')
var vfile = require('vfile')
```

I can run `mdast-util-to-nlcst` directly in one of two ways: either (redundantly?) on the output of `remark-retext` in Option 1, or by bypassing `remark-retext` entirely in Option 2.
```js
// Option 1
var tree_from_md = unified().use(parse).use(remark2retext, english.Parser).parse('## Hello World')
var nlcst_from_md = toNlcst(tree_from_md, vfile('## Hello World'), english.Parser)

// Option 2
var tree_from_direct_md = unified().use(parse).parse('## Hello World')
var nlcst_from_direct_md = toNlcst(tree_from_direct_md, vfile('## Hello World'), english.Parser)
```

In either case, the first child of the tree is of type `ParagraphNode`, which indicates to me that this tree is in fact `nlcst`. However, the `mdast-util-to-nlcst` documentation suggests that I instead use `remark-retext`. Is there in fact a way to retrieve `nlcst` trees from `remark-retext` without directly calling `mdast-util-to-nlcst`?

Happy holidays and many thanks to anyone who might happen to see this.

**Comment by wooorm** [comment, marked as answer]
Hi there!

The `.parse` function only performs the parsing stage. `.run` and `.runSync` can be used to perform the run stage (running plugins).
[More info available here](https://github.com/unifiedjs/unified#description).

And it’s good to see them as different processors:

```js
var processor = unified().use(parse).use(remark2retext, english.Parser)

var tree_from_md = processor.runSync(processor.parse(doc))
```

**Reply by tvquizphd** [reply]
Thanks for such a quick response! This makes a lot of sense: I should have read the docs more carefully to understand the parse, run, and stringify methods of a processor. I think it's pretty clear now. Thanks again.
